### ISSUE TITLE
Test test_all_pyflakes_messages_have_flake8_codes_assigned failing

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#361]
In GitLab by @mcepl on May 4, 2020, 13:31

*Please describe how you installed Flake8*

When running the test suite while building the official packages for openSUSE
```
[   33s] + PYTHONPATH=:/home/abuild/rpmbuild/BUILDROOT/python-flake8-3.7.9-0.x86_64/usr/lib/python3.8/site-packages
[   33s] + PYTHONDONTWRITEBYTECODE=1
[   33s] + py.test-3.8 --ignore=_build.python2 --ignore=_build.python3 --ignore=_build.pypy3 -v tests
[   34s] ============================= test session starts ==============================
[   34s] platform linux -- Python 3.8.2, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
[   34s] cachedir: .pytest_cache
[   34s] rootdir: /home/abuild/rpmbuild/BUILD/flake8-3.7.9, inifile: pytest.ini
[   34s] collecting ... collected 451 items

[...]

[   36s] =================================== FAILURES ===================================
[   36s] ____________ test_all_pyflakes_messages_have_flake8_codes_assigned _____________
[   36s] 
[   36s]     def test_all_pyflakes_messages_have_flake8_codes_assigned():
[   36s]         """Verify all PyFlakes messages have error codes assigned."""
[   36s]         messages = {
[   36s]             name
[   36s]             for name, obj in vars(pyflakes.messages).items()
[   36s]             if name[0].isupper() and obj.message
[   36s]         }
[   36s] >       assert messages == set(pyflakes_shim.FLAKE8_PYFLAKES_CODES)
[   36s] E       AssertionError: assert {'AssertTuple...NotLast', ...} == {'AssertTuple...NotLast', ...}
[   36s] E         Extra items in the left set:
[   36s] E         'PercentFormatExpectedMapping'
[   36s] E         'StringDotFormatInvalidFormat'
[   36s] E         'StringDotFormatMissingArgument'
[   36s] E         'PercentFormatExpectedSequence'
[   36s] E         'StringDotFormatExtraNamedArguments'
[   36s] E         'PercentFormatStarRequiresSequence'...
[   36s] E         
[   36s] E         ...Full output truncated (62 lines hidden), use '-vv' to show
[   36s] 
[   36s] tests/unit/test_pyflakes_codes.py:16: AssertionError
[   36s] ======================== 1 failed, 450 passed in 2.37s =========================
```
**Note**: Some *nix distributions patch Flake8 arbitrarily to accommodate incompatible software versions. If you're on one of those distributions, your issue may be closed and you will be asked to open an issue with your distribution package maintainers instead.

This has been reproduced with all distribution patches specifically switched off.

*Please provide the exact, unmodified output of `flake8 --bug-report`*:
```
[    4s] + python3 /home/abuild/rpmbuild/BUILDROOT/python-flake8-3.7.9-0.x86_64/usr/bin/flake8-3.8 --bug-report
[    4s] {
[    4s]   "dependencies": [
[    4s]     {
[    4s]       "dependency": "entrypoints",
[    4s]       "version": "0.3"
[    4s]     }
[    4s]   ],
[    4s]   "platform": {
[    4s]     "python_implementation": "CPython",
[    4s]     "python_version": "3.8.2",
[    4s]     "system": "Linux"
[    4s]   },
[    4s]   "plugins": [
[    4s]     {
[    4s]       "is_local": false,
[    4s]       "plugin": "mccabe",
[    4s]       "version": "0.6.1"
[    4s]     },
[    4s]     {
[    4s]       "is_local": false,
[    4s]       "plugin": "pycodestyle",
[    4s]       "version": "2.5.0"
[    4s]     },
[    4s]     {
[    4s]       "is_local": false,
[    4s]       "plugin": "pyflakes",
[    4s]       "version": "2.2.0"
[    4s]     }
[    4s]   ],
[    4s]   "version": "3.7.9"
[    4s] }
```
Complete [log with all details](/uploads/eaf7a21eb4b62927be87b1e97921bf36/_log.txt).

**Comment by asottile** [comment]
In GitLab by @asottile on May 4, 2020, 13:33

this is fixed on master and 3.8.0a2, you're using an old version of flake8 (3.7.9) with an incompatible pyflakes (2.2.0)

from the issue template:

> **Note**: Some \*nix distributions patch Flake8 arbitrarily to accommodate incompatible software versions. If you're on one of those distributions, your issue may be closed and you will be asked to open an issue with your distribution package maintainers instead.

**Comment by asottile** [comment]
In GitLab by @asottile on May 4, 2020, 13:33

closed
